integration_test/otp_test.dart

```diff
@@ -1,165 +1,240 @@
-    });
-
-    appTest('Swap slots', (WidgetTester tester) async {
-      await tester.tap(find.byKey(otpAppDrawer).hitTestable());
-      await tester.shortWait();
-
-      /// TODO:  verify "Slot 1 is configured"
-      /// TODO:  verify "Slot 2 is configured"
-
-      // taps swap
-      await tester.tapSwapSlotsButton();
-      await tester.tap(find.byKey(swapButton).hitTestable());
-      await tester.shortWait();
-
-      /// TODO:  verify "Slot 1 is configured"
-      /// TODO:  verify "Slot 2 is configured"
-    });
-
-    appTest('Delete Credentials', (WidgetTester tester) async {
-      await tester.tap(find.byKey(otpAppDrawer).hitTestable());
-      await tester.shortWait();
-
-      /// TODO:  verify "Slot 1 is configured"
-      /// TODO:  verify "Slot 2 is configured"
-
-      await tester.openSlotMenu(SlotId.one);
-      await tester.tap(find.byKey(deleteAction).hitTestable());
-      await tester.shortWait();
-      await tester.tap(find.byKey(deleteButton).hitTestable());
-
-      /// TODO:  wait for any toasts to be gone
-      await tester.ultraLongWait();
-      var closeFinder = find.byKey(closeButton);
-      if (closeFinder.evaluate().isNotEmpty) {
-        // close the view
-        await tester.tap(closeFinder);
-        await tester.shortWait();
-      }
-
-      // we need to right click on slot 2
-      await tester.openSlotMenu(SlotId.two);
-      await tester.tap(find.byKey(deleteAction).hitTestable());
-      await tester.shortWait();
-      await tester.tap(find.byKey(deleteButton).hitTestable());
-      await tester.shortWait();
-
-      /// TODO:  verify "Slot 1 is empty"
-      /// TODO:  verify "Slot 2 is empty"
-    });
-  });
+  appGroup(
+    'OTP',
+    (params) {
+      testKey('Swap slots', params, ($, data) async {
+        await $.navigate(Section.slots);
+
+        // Ensure slot 2 is empty
+        final state = $.read(otpStateProvider(data.node.path)).value!;
+        expect(
+          state.slot2Configured,
+          isFalse,
+          reason: 'Slot 2 should not be configured initially',
+        );
+
+        if (!state.slot1Configured) {
+          await $
+              .read(otpStateProvider(data.node.path).notifier)
+              .configureSlot(
+                SlotId.one,
+                configuration: SlotConfiguration.chalresp(key: 'cafed00d'),
+              );
+          await $.pumpAndSettle();
+        }
+
+        // Swap slots
+        await $.viewAction(swapSlots);
+        await $(swapButton).tap();
+        await $.condition(
+          () =>
+              $.read(otpStateProvider(data.node.path)).value?.slot2Configured ==
+              true,
+        );
+        expect(
+          $.read(otpStateProvider(data.node.path)).value?.slot1Configured,
+          isFalse,
+        );
+
+        if (state.slot1Configured) {
+          // Slot 1 was pre-programmed, swap back
+          await $.read(otpStateProvider(data.node.path).notifier).swapSlots();
+        } else {
```

Comment:
Since we perform the swap above, Slot 1 will always be empty, so deleting it will fail. We should probably either delete Slot 2 here or always perform the swap back and only delete Slot 1 if it was not pre-programmed.